### PR TITLE
Add page size limit

### PR DESCRIPTION
**Pull request by pulpn0ir**
Allows setting a page size limit when creating a `DatabaseContext` to prevent server overloads.

If the requested page size exceeds the limit, the response page is capped to the defined limit. The default value is `nil`, allowing for an unlimited page size.

This PR relates to https://github.com/vapor/fluent/pull/715

### CODE REVIEW

**Comment by siemensikkema** [comment]
This PR is the continuation of #406

File: Sources/FluentKit/Database/Database.swift
Comment (calebkleveter) [inline Sources/FluentKit/Database/Database.swift:49]: The name is a little ambiguous. Something like `pageSizeLimit` would be more self-explanatory.

Comment (siemensikkema) [reply Sources/FluentKit/Database/Database.swift:49]: @calebkleveter I can get on board with that. @pulpn0ir what about you?

Comment (pulpn0ir) [reply Sources/FluentKit/Database/Database.swift]: Fully agree. I updated both PRs.

**Review by 0xTim — APPROVED** [review]

**Comment by Joannis** [comment]
@0xTim any reason this wasn't merged (yet)?

**Comment by 0xTim** [comment]
@siemensikkema I was waiting for you to give the final OK for this, this good to merge? Don't merge just yet if so, just approve. I'll be sorting out the release bot in a couple of hours so we can wait for that

**Comment by Joannis** [comment]
After a small discussion with @0xTim it seems he wants at least @gwynne or @siemensikkema to look into this as well.

**Review by gwynne — CHANGES_REQUESTED** [review]
Looks mostly fine to me, but I'd like the unit tests to include a test for a page other than the first (in particular, a test that pages after the first are at the correct offset relative to the given limits).

**Comment by siemensikkema** [comment]
@gwynne good idea. I changed the test slightly to ensure this is covered. I chose to reduce the page limit so I could request page 2 while still getting a full page. The other choice I could have made would have been to add one more row to the test output.

**Pull request merged**
